pythonbits/scene.py

```diff
@@ -1,81 +1,383 @@
+
+    Raise APIError in case of connection or file system error.
+    """
+    # Cover the following cases:
+    #   - path is directory (season) and scene released directory
+    #   - path is directory and scene released individual file (episodes)
+    #   - path is file and scene released individual file
+    #   - path is file and scene released directory
+    _, files = get_details(path)
+
+    def default(filepath):
+        return {'release_name': None, 'filename': os.path.basename(filepath), 'size': None, 'crc': None}
+
+    if files:
+        # Scene release is the same as this release (season pack or single episode)
+        yield from files.values()
+
+    elif not _isdir(path):
+        # path is episode and scene release is season pack
+        log.debug('Looking for episode from season pack')
+        guess = _guessit(path)
+        # Try to find whole season if we searched for single episode
+        if 'release_group' in guess and 'season' in guess and 'episode' in guess:
+            guess.pop('episode', None)
+            results = search(guess)
+            # If we find the exact season, pick episode info
+            if len(results) == 1:
+                _, files = get_details(results[0])
+                yield files.get(os.path.basename(path), default(path))
+            else:
+                yield default(path)
+        else:
+            yield default(path)
 
-    if results:
-        print('Found srrDB results for filename:')
-        print("\t" + "\n".join(r['release'] for r in results))
     else:
-        print('No results found in srrDB for query "{}"'.format(query))
+        log.debug('Making season pack from individual episodes')
+        # path is season and scene released single episodes
+        try:
+            files = _os_listdir(path)
+        except OSError as e:
+            raise APIError('%s: %s' % (path, e.strerror))
+        else:
+            log.debug('Found files:', files)
+            for filename in sorted(files):
+                _, files = get_details(filename)
+                yield files.get(filename, default(filename))
+
+
+@functools.lru_cache()
+def _guessit(filepath):
+    release_name = os.path.basename(filepath)
+    if release_name.islower():
+        match = re.search(r'^([a-z0-9]+)-(.+)\.([a-z0-9]{1,3})$', release_name)
+        if match:
+            release_name = '%s-%s.%s' % (match.group(2), match.group(1), match.group(3))
+    return dict(guessit.guessit(release_name))
+
+
+@functools.lru_cache()
+def _get(path):
+    url = srrdb + path
+    log.debug('Requesting URL: {}', url)
+    try:
+        return requests.get(url)
+    except requests.RequestException as e:
+        log.debug('Request failed: {}', e)
+        raise APIError(e)
+
+
+def _get_basename_noext(path):
+    # os.path.splitext() removes anything after the rightmost "." which is no good for
+    # scene release names.
+    return re.sub(r'\.[a-z0-9]{1,3}$$', '', os.path.basename(path))
```

Comment:
Why this instead of something like `os.path.splitext(os.path.basename(path))[0]`?

Reply:
```python
>>> os.path.splitext(os.path.basename("The.Title.2000.x265-GRP"))
('The.Title.2000', '.x265-GRP')
```